osi_logicallane.proto

```diff
@@ -0,0 +1,471 @@
+    // A point is "after" the reference line, if its s coordinate is > sEnd.
+    //
+    // To describe points that are not directly on the polyline, a T
+    // coordinate is added. T is the signed 2D distance (i.e. hypot(A.X-B.X,
+    // A.Y-B.Y), if A and B are the two points) between the point to describe
+    // and the nearest point on the polyline (this point might either be on a
+    // line segment or at an edge between two line segments). The distance is
+    // positive if the point is left of the polyline (in definition direction),
+    // negative if it is right of it. Note that the "nearest point on the
+    // polyline" is calculated in 3D, in order to choose the correct point for
+    // 3D curves (think reference lines for roads in parking decks). If there
+    // are several "nearest points", the one with the smallest S coordinate on
+    // the polyline is chosen.
+    //
+    // Sometimes an angle to a reference line is needed. This shall be defined
+    // as follows:
+    // First the nearest point on the polyline is determined, as described
+    // above. If this point is on a line segment, then the angle is calculated
+    // relative to the line segment on which the reference point lays.
+    // If the nearest point is at the edge between line segments, then the
+    // angle of the following line shall be chosen.
+    //
+    // Notes on OpenDRIVE compatibility:
+    // Ideally, one would want the polyline to be fully compatible with
+    // OpenDRIVE, so that calculations done for OpenDRIVE directly match those
+    // in OSI. There are a few difficulties with this:
+    // - The T coordinate is nearly the same as for OpenDRIVE, but
+    //   unfortunately not perfectly. In OpenDRIVE, if the road is tilted using
+    //   superElevation, then the t coordinate system is tilted along, so the T
+    //   coordinate is no longer calculated in the XY plane (as proposed for
+    //   OSI). It doesn't seem feasable to implement the same tilting for OSI,
+    //   so simulation tools will have to consider superElevation and convert
+    //   the T coordinate accordingly.
+    // - The angle will not be perfectly the same, due to the use of line
+    //   segments in OSI, and curves in OpenDRIVE. In the authors opinion, the
+    //   difference will be negligible if the poly_line is suitably sampled.
+    //
+    // Notes on design decisions:
+    // - The S coordinate is included directly, both for OpenDRIVE
+    //   compatibility, and to speed up calculations.
+    // - The rules on S coordinates (e.g. the calculation in 2D space) are
+    //   there to ensure OpenDRIVE compatibility.
+    // - The rules on T coordinates are there to ensure OpenDRIVE compatibility
+    //   for lanes without superelevation, and to make it easier to convert
+    //   between OSI and OpenDRIVE in case superelevation is present.
+    //
+    // TODO Describe needed precision.
```

Comment:
Is a minimum of points per length/curverture required in order t prevent errors regarding interpolation and t coordinate assignment?

Reply:
Addressed in the latest commit.